common/src/types/wallet/balances.rs

```diff
@@ -82,24 +82,22 @@ impl Wallet {
             return Ok(());
         }
 
-        // Otherwise, add the balance
-        if self.balances.len() < MAX_BALANCES {
-            self.balances.insert(balance.mint.clone(), balance);
-            return Ok(());
+        if let Some(index) = self.find_first_replaceable_balance() {
+            self.balances.insert_at_index(index, balance.mint.clone(), balance);
```

Comment:
KeyedList's `[insert_at_index](https://github.com/renegade-fi/renegade/blob/joey/balance-append-logic/common/src/keyed_list.rs#L155)` method inserts an element at position index within the vector, shifting all elements after it to the right. My understanding was the behavior we want is to replace the zero balance (and zero order).

Reply:
Should we use `replace_at_index` instead if this is indeed the correct behavior?

Reply:
Actually, I remember you mentioning that balances must maintain their order, so we cannot overwrite/replace for balances.